src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
This test is definitely too large for jQuery.

I'd share your concerns a year ago but now latest stable versions of Chrome & Firefox support symbols so I doubt any code using jQuery 3.0.0 (note that we're not releasing tomorrow, it'll take at least 1-2 months) will rely on behavior in browsers without symbols, rather the opposite.

I don't have a definite opinion about removing the `Symbol.iterator` check.